IA-2047 IA-2282 IA-2252 Completeness stats improvement

olethanh opened the pull request:
Make planning filter work properly in the Completeness stats page, it now filter on the assigned Form and OrgUnit


Related JIRA tickets : IA-2047 IA-2282 IA-2252

## Self proofreading checklist

- [ ] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- Make planning filter work properly in the Completeness stats page, it now filter on the assigned Form and OrgUnit
- fix crash when navigating hierarchy
- Seed better data for planning and completeness so we have data for testing

## How to test

Seed a iaso database via the command or create a whole planning with proper teams and assignment
Go to the completeness stat page

madewulf commented on iaso/dhis2/derived_instance_generator.py:
```diff
@@ -106,6 +112,27 @@ def generate_instances(project, cvs_form, cvs_stat_mapping_version, period):
     return progress
 
 
+def condition_to_q(condition):
```
This looks awesome, and something that we would probably benefit to expose in the usual search for submissions.

olethanh committed:
IA-2047 Improve seed data for completeness

so it's more interesting by default

Add probability to not have submssion for a period
Remove unkown unit tyype if we don't improve any
Properly set OrgUnitType on Form

olethanh committed:
move genUrl from plugins polio to iaso

olethanh committed:
Move to typescript

olethanh committed:
move genUrl from plugins polio to iaso. Convert to typescript

olethanh committed:
IA-2282 Completeness: fix crash when navigating hierarchy

When settings a validation status and going to children or parent page via the action in the Table
it was 404ing because the parameters where not in the correct order
The code now use react-router to generate the new url like we did in polio
this use the pattern in the routing and ensure we use the correct order for params

olethanh committed:
Clean and simplify code

olethanh committed:
Better doc string

olethanh committed:
Remove multi on planning filter as it doesn't work

olethanh committed:
Reset page when going to parent

olethanh committed:
IA-2047 provide more interesting data in seed planning

for completenessStat but also in general

olethanh committed:
IA-2252 Completeness stats: planning filter follow proper assignement

so we now display stat for the planning

olethanh committed:
adapt tests

Corona79 commented:
Good work

olethanh committed:
adapt test bis

missed one

olethanh commented:
forgot to mention but we now also delete the `unkown_` type after a dhis2 import if there is no orgunit with this type. This is related to the previous change in dhis2 import where we deduced the type from the orgunitlevels api in DHIS2

quang-le reviewed (changes requested):
If you select a form and a planning not linked to that form, you get a 500:

https://github.com/BLSQ/iaso/assets/38907762/e7b2492c-d905-45cd-a598-f0792a8b4732

beygorghor committed:
Merge branch 'main' into IA-2047_improve_seed_data_completeness

beygorghor committed:
Merge branch 'main' into IA-2047_improve_seed_data_completeness
